File: src/main/java/model/equipment/shield/LightShield.java
package model.equipment.shield;

import model.support.allEnums.DmgType;
import model.support.allEnums.EquipmentType;
import model.support.allEnums.ShieldType;

import static model.support.Constants.FIVE;

public class LightShield extends Shield {
    public LightShield(double weight, int quality) {
        super(EquipmentType.SHIELD, ShieldType.LIGHT_SHIELD, weight, quality);
        price = quality / FIVE - weight * 2;
        setDefense();
    }

    private void setDefense() {
        defense.put(DmgType.BLUNT, quality * 0.20);
        defense.put(DmgType.CUT, quality * 0.20);
        defense.put(DmgType.SLASH, quality * 0.20);
        defense.put(DmgType.PIERCING, quality * 0.20);
        defense.put(DmgType.CRUSH, quality * 0.20);
    }
}

File: src/main/java/model/equipment/shield/LightShield.kt
package model.equipment.shield

import model.support.Constants.FIVE
import model.support.enums.EquipmentType.SHIELD
import model.support.enums.ShieldType.LIGHT_SHIELD
import model.support.enums.Specification

class LightShield(
        weight: Double,
        quality: Int
) : Shield(
        SHIELD,
        LIGHT_SHIELD,
        weight,
        quality,
        quality / FIVE - weight * 2
) {
    init {
        with(defense) {
            put(Specification.BLUNT, quality * 0.20)
            put(Specification.CUT, quality * 0.20)
            put(Specification.SLASH, quality * 0.20)
            put(Specification.PIERCING, quality * 0.20)
            put(Specification.CRUSH, quality * 0.20)
        }
    }
}
